node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java

```diff
@@ -63,44 +66,94 @@ public StorageMaintainer(DockerOperations dockerOperations, ProcessExecuter proc
 
         Dimensions dimensions = new Dimensions.Builder().add("role", "docker").build();
         numberOfNodeAdminMaintenanceFails = metricReceiver.declareCounter(MetricReceiverWrapper.APPLICATION_DOCKER, dimensions, "nodes.maintenance.fails");
+        numberOfCoredumpsOnHost = metricReceiver.declareGauge(MetricReceiverWrapper.APPLICATION_DOCKER, dimensions, "nodes.coredumps");
     }
 
     public void writeMetricsConfig(ContainerName containerName, NodeSpec node) {
-        final Path yamasAgentFolder = environment.pathInNodeAdminFromPathInNode(
-                containerName, Paths.get("/etc/yamas-agent/"));
+        List<SecretAgentCheckConfig> configs = new ArrayList<>();
 
+        // vespa
         Path vespaCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_vespa");
-        SecretAgentCheckConfig vespaSchedule = new SecretAgentCheckConfig("vespa", 60, vespaCheckPath, "all")
-                .withTag("parentHostname", environment.getParentHostHostname());
+        SecretAgentCheckConfig vespaSchedule = new SecretAgentCheckConfig("vespa", 60, vespaCheckPath, "all");
+        configs.add(annotatedCheck(node, vespaSchedule));
 
+        // vespa-health
+        Path vespaHealthCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_vespa_health");
+        SecretAgentCheckConfig vespaHealthSchedule = new SecretAgentCheckConfig("vespa", 60, vespaHealthCheckPath, "all");
+        configs.add(annotatedCheck(node, vespaHealthSchedule));
+
+        // host-life
         Path hostLifeCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_host_life");
-        SecretAgentCheckConfig hostLifeSchedule = new SecretAgentCheckConfig("host-life", 60, hostLifeCheckPath)
-                .withTag("namespace", "Vespa")
+        SecretAgentCheckConfig hostLifeSchedule = new SecretAgentCheckConfig("host-life", 60, hostLifeCheckPath);
+        configs.add(annotatedCheck(node, hostLifeSchedule));
+
+        // ntp
+        Path ntpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_ntp");
+        SecretAgentCheckConfig ntpSchedule = new SecretAgentCheckConfig("host-life", 60, ntpCheckPath);
+        configs.add(annotatedCheck(node, ntpSchedule));
+
+        // coredumps (except for the done coredumps which is handled by the host)
+        Path coredumpCheckPath = environment.pathInNodeUnderVespaHome("libexec/yms/yms_check_coredumps");
+        SecretAgentCheckConfig coredumpSchedule = new SecretAgentCheckConfig("system-coredumps-processing", 300,
+                coredumpCheckPath, "--application", "system-coredumps-processing", "--lastmin",
+                "129600", "--crit", "1", "--coredir", "/home/y/var/crash/processing");
```

Comment:
`/home/y` -> `environment.pathInNodeUnderVespaHome()`

Reply:
The check is ran in the container right?

Reply:
My underastanding was that pathInNodeUnderVespaHome is used to find the path when on the host but when you are in the container you would refer to the path directly (at least not the same path as when at the host)